src/Functions/errorCode.cpp

```diff
@@ -0,0 +1,87 @@
+#include <Functions/castTypeToEither.h>
+#include <Functions/IFunctionImpl.h>
+#include <Functions/FunctionFactory.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Columns/ColumnsNumber.h>
+#include <Columns/ColumnString.h>
+#include <Core/Field.h>
+#include <unordered_map>
+
+#if !defined(ARCADIA_BUILD)
+#    include <Common/config_version.h>
+#endif
+
+extern std::unordered_map<int, const char *> auto_error_codes;
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+/** errorCode() - returns the macro name for the error code.
+  */
+class FunctionErrorCode : public IFunction
+{
+public:
+    static constexpr auto name = "errorCode";
+    static FunctionPtr create(const Context &)
+    {
+        return std::make_shared<FunctionErrorCode>();
+    }
+
+    String getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 1; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & types) const override
+    {
+        if (!isNumber(types.at(0)))
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "The argument of function {} must have simple numeric type, possibly Nullable", name);
+
+        return std::make_shared<DataTypeString>();
+    }
+
+    ColumnPtr executeImpl(ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        const ColumnPtr input_column = arguments[0].column;
+        auto col_res = ColumnString::create();
+
+        if (!castTypeToEither<
+            ColumnUInt8, ColumnUInt16, ColumnUInt32, ColumnUInt64,
+            ColumnInt8, ColumnInt16, ColumnInt32, ColumnInt64,
+            ColumnFloat32, ColumnFloat64>(
+            input_column.get(), [&](const auto & column)
+            {
+                for (size_t i = 0; i < input_rows_count; ++i)
+                {
+                    const Int64 error_code = column.getInt(i);
+                    std::string macro_name;
```

Comment:
Excessive allocation in a loop + extra call to strlen.
You can use std::string_view for values... but simple switch/case with string literals is Ok.